scan/src/components/oracle-script/OracleScriptExecute.re

```diff
@@ -0,0 +1,213 @@
+
+  let input =
+    style([
+      width(`percent(100.)),
+      background(white),
+      paddingLeft(`px(20)),
+      fontSize(`px(12)),
+      fontWeight(`num(500)),
+      outline(`px(1), `none, white),
+      height(`px(40)),
+      borderRadius(`px(4)),
+      boxShadow(
+        Shadow.box(~inset=true, ~x=`zero, ~y=`zero, ~blur=`px(4), Css.rgba(0, 0, 0, 0.1)),
+      ),
+    ]);
+
+  let buttonContainer = style([display(`flex), flexDirection(`row), alignItems(`center)]);
+
+  let button = isLoading =>
+    style([
+      width(`px(isLoading ? 150 : 110)),
+      backgroundColor(isLoading ? Colors.blueGray3 : Colors.green2),
+      borderRadius(`px(6)),
+      fontSize(`px(12)),
+      fontWeight(`num(600)),
+      color(isLoading ? Colors.blueGray7 : Colors.green7),
+      cursor(isLoading ? `auto : `pointer),
+      padding2(~v=Css.px(10), ~h=Css.px(10)),
+      whiteSpace(`nowrap),
+      outline(`px(0), `none, white),
+      boxShadow(
+        isLoading
+          ? `none : Shadow.box(~x=`zero, ~y=`px(2), ~blur=`px(4), Css.rgba(0, 0, 0, 0.1)),
+      ),
+      border(`zero, `solid, Colors.white),
+    ]);
+
+  let hFlex = style([display(`flex), flexDirection(`row)]);
+
+  let withWH = (w, h) =>
+    style([
+      width(w),
+      height(h),
+      display(`flex),
+      justifyContent(`center),
+      alignItems(`center),
+    ]);
+
+  let resultWrapper = (w, h, overflow_choice) =>
+    style([
+      width(w),
+      height(h),
+      display(`flex),
+      flexDirection(`column),
+      justifyContent(`center),
+      backgroundColor(Colors.white),
+      borderRadius(`px(4)),
+      overflow(overflow_choice),
+    ]);
+};
+
+let parameterInput = (name, placeholder, index, setCalldataArr) => {
+  <div className=Styles.listContainer key=name>
+    <Text value=name size=Text.Md color=Colors.gray6 />
+    <VSpacing size=Spacing.xs />
+    <input
+      className=Styles.input
+      type_="text"
+      onChange={event => {
+        let newVal = ReactEvent.Form.target(event)##value;
+        setCalldataArr(prev => {
+          prev->Belt_Array.set(index, newVal);
+          prev;
+        });
+      }}
+      placeholder
+    />
+  </div>;
+};
+
+type result_t =
+  | Nothing
+  | Loading
+  | Error(string)
+  | Success(string);
+
+let loadingRender = (wDiv, wImg, h) => {
+  <div className={Styles.withWH(wDiv, h)}>
+    <img src=Images.loadingCircles className={Styles.withWH(wImg, h)} />
+  </div>;
+};
+
+let resultRender = result => {
+  switch (result) {
+  | Nothing => React.null
+  | Loading =>
+    <>
+      <VSpacing size=Spacing.xl />
+      {loadingRender(`percent(100.), `px(104), `px(30))}
+      <VSpacing size=Spacing.lg />
+    </>
+  | Error(err) =>
+    <>
+      <VSpacing size=Spacing.lg />
+      <div className={Styles.resultWrapper(`percent(100.), `px(90), `scroll)}>
+        <Text value=err />
+      </div>
+    </>
+  | Success(output) =>
+    <>
+      <VSpacing size=Spacing.lg />
```

Comment:
This would be a new ui  (report loading tube)? (Add todo and remove data source code)